tests/pytests/unit/fileserver/test_s3fs.py

```diff
@@ -1,25 +1,142 @@
+import os
+
+import boto3
 import pytest
+import yaml
+
+# moto must be imported before boto3
+from moto import mock_s3
```

Comment:
Yes really. I'm not totally positive, but their [docs](https://docs.getmoto.org/en/latest/docs/getting_started.html#how-do-i-avoid-tests-from-mutating-my-real-infrastructure) state it pretty clearly. I'm guessing it's because there are either some cases or versions or something that run authentication initialization during the import, so importing moto first ensures we can always beat it to the punch, so to speak.

I generally found moto very finicky to work with during the implementation of these tests. Much care is needed to avoid nesting scopes and such. I guess that's just mocks though.